src/TorchSharp/DataLoader.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+using TorchSharp.Utils;
+
+namespace TorchSharp
+{
+    public static partial class torch
+    {
+        public static partial class utils
+        {
+            public static partial class data
+            {
+                public class DataLoader : IEnumerable<Dictionary<string, Tensor>>, IDisposable
```

Comment:
Please add doc comments to all classes and (at least) all non-private members. You can copy information from the Pytorch docs, if there is any.